server/src-lib/Hasura/RQL/IR/BoolExp.hs

```diff
@@ -244,6 +244,11 @@ data OpExpG (b :: BackendType) a
   | ASIMILAR !a -- similar, regex
   | ANSIMILAR !a-- not similar, regex
 
+  | APOSIXCS !a -- match POSIX case sensitive, regex
+  | APOSIXCI !a -- match POSIX case insensitive, regex
+  | ANPOSIXCS !a-- dont match POSIX case sensitive, regex
+  | ANPOSIXCI !a-- dont match POSIX case insensitive, regex
```

Comment:
We are in the process of reorganizing and generalizing a bunch of code in the RQL modules. However, it's not necessary for this PR to participate in that process. But let's add a comment that documents what would need to be done in the future here.
```suggestion
  -- Now that in the RQL code we've started to take a "trees that grow"
  -- approach (see PR #6003), we may eventually want to move these
  -- recently added constructors, which correspond to newly supported
  -- Postgres operators, to the backend-specific extensions of this type.
  | APOSIXCS !a -- match POSIX case sensitive, regex
  | APOSIXCI !a -- match POSIX case insensitive, regex
  | ANPOSIXCS !a-- dont match POSIX case sensitive, regex
  | ANPOSIXCI !a-- dont match POSIX case insensitive, regex
```